### PR TITLE
Replacing header nav partial with header component

### PR DESCRIPTION
**Pull request by hudajkhan**
**Context**
Relates to having tests pass with Blacklight 8.3 and using a Spotlight HeaderComponent instead of shared/_header_navbar.  The Spotlight header is almost exactly the same as the default Blacklight header except it does not include the search bar (which is elsewhere on the page). 
Refer to https://github.com/projectblacklight/spotlight/issues/3047 and this comment which includes a checklist of tests to fix: https://github.com/projectblacklight/spotlight/issues/3047#issuecomment-2359456286

For 8.3, the following tests are failing:
 - rspec ./spec/views/shared/_header_navbar.html.erb_spec.rb:8 # shared/_header_navbar has nav links
 - rspec ./spec/views/spotlight/pages/show.html.erb_spec.rb:46 # spotlight/pages/show when rendering with layout does not double-escape HTML entities in the HTML title
 - rspec ./spec/views/spotlight/pages/show.html.erb_spec.rb:50 # spotlight/pages/show when rendering with layout includes analytics reporting

**What this pull request does**
- Removes the header_navbar partial.
- Creates a Spotlight HeaderComponent whose partial displays on the top bar. 
- Adds the Spotlight Header Component to the configuration as the header component.
- Adds blacklight configuration information for the show html rspec to pass. 

**How to test**
These tests should still pass in Blacklight 7.

How to test with Blacklight 8: Generate test app (.internal_test_app). Update the internal test app's Gemfile to use Blacklight 8.3. Run bundle install for the internal test app. Then run rspec ./spec/views/spotlight/pages/show.html.erb_spec.rb:46 and rspec ./spec/views/spotlight/pages/show.html.erb_spec.rb:50.

### CODE REVIEW

**Comment by cbeer** [comment]
I wonder if there's a backwards-compatible way to introduce this change (that won't require downstream applications to add the `header_component` configuration).

Maybe we could keep the header_navbar partial?

```
<% if blacklight_config.header_component.present? && blacklight_config.header_component != Blacklight::HeaderComponent %>
<%= render blacklight_config.header_component.new(blacklight_config: blacklight_config) %>
<% else %>
<%# TODO: deprecation warning to encourage downstream to add the config? %>
<%= render Spotlight::HeaderComponent.new(blacklight_config: blacklight_config) %>
<% end %>
```

**Comment by hudajkhan** [comment]
@cbeer I brought the partial back and its associated test and set up with a deprecation warning.

File: spec/views/shared/_header_navbar.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+describe 'shared/_header_navbar', type: :view do
```
Comment (jcoyne) [inline spec/views/shared/_header_navbar.html.erb_spec.rb]:
```suggestion
RSpec.describe 'shared/_header_navbar', type: :view do
```

File: spec/views/shared/_header_navbar.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  before do
+    allow(view).to receive(:blacklight_config).and_return(Blacklight.default_configuration)
+    stub_template 'shared/_user_util_links.html.erb' => 'links'
+  end
+
+  it 'has nav links' do
+    render
+    expect(rendered).to have_selector '#user-util-collapse', text: 'links'
```
Comment (jcoyne) [inline spec/views/shared/_header_navbar.html.erb_spec.rb]: Do we need to have a view test at all?  View tests are deprecated in rails.  Instead can you use a request test, feature test, or a component test?

File: app/views/shared/_header_navbar.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<% if blacklight_config.header_component.present? && blacklight_config.header_component != Blacklight::HeaderComponent %>
+  <%= render blacklight_config.header_component.new(blacklight_config: blacklight_config) %>
+<% else %>
+  <%= render Spotlight::HeaderComponent.new(blacklight_config: blacklight_config) %>
+  <% ActiveSupport::Deprecation.warn("_header_navbar.html.erb will be deprecated in future versions. Please use Blaclight configuration to specify your HeaderComponent.") %>
```
Comment (jcoyne) [inline app/views/shared/_header_navbar.html.erb]:
```suggestion
  <% ActiveSupport::Deprecation.warn("_header_navbar.html.erb will be deprecated in future versions. Please use Blacklight configuration to specify your HeaderComponent.") %> 
```

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by hudajkhan** [comment]
@jcoyne I addressed your suggestions.  Rubocop said removing the "freeze" line at the beginning of the file was wrong (for the feature test version)

**Comment by jcoyne** [comment]
@hudajkhan needs a rebase

File: app/views/shared/_header_navbar.html.erb [hunk 4]
```diff
@@ -1,12 +1,6 @@
...
-    <div class="collapse navbar-collapse justify-content-md-end" id="user-util-collapse">
-      <%= render partial: 'shared/user_util_links' %>
-    </div>
-  </div>
-</nav>
+<% if blacklight_config.header_component.present? && blacklight_config.header_component != Blacklight::HeaderComponent %>
+  <%= render blacklight_config.header_component.new(blacklight_config: blacklight_config) %>
+<% else %>
+  <%= render Spotlight::HeaderComponent.new(blacklight_config: blacklight_config) %>
+  <% ActiveSupport::Deprecation.new('5.0', 'Spotlight').warn("_header_navbar.html.erb will be deprecated in future versions. Please use Blacklight configuration to specify your HeaderComponent.") %>
```
Comment (cbeer) [inline app/views/shared/_header_navbar.html.erb:5]:
```suggestion
  <% Spotlight.deprecator.warn("_header_navbar.html.erb will be deprecated in future versions. Please use Blacklight configuration to specify your HeaderComponent.") %>
```

Comment (hudajkhan) [reply app/views/shared/_header_navbar.html.erb]: Changes, thanks

**Review by jcoyne — APPROVED** [review]

**Pull request merged**